test/unit/host-config.spec.ts

```diff
@@ -0,0 +1,701 @@
+					log_info: 'on',
+					log: 'stderr',
+					daemon: 'off',
+					redirector: 'iptables',
+				},
+			});
+		});
+
+		it('parses to empty base config while all values are invalid', () => {
+			const redsocksConfStr = stripIndent`
+                base {
+                    log_debug = on;
+                    log_info = off;
+                    log = stdout;
+                    daemon = on;
+                    redirector = foo;
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({});
+		});
+
+		it('parses to empty base config while some values are invalid', () => {
+			// Only `redirector` is invalid
+			const redsocksConfStr = stripIndent`
+                base {
+                    log_debug = off;
+                    log_info = on;
+                    log = stderr;
+                    daemon = off;
+                    redirector = foo;
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({});
+		});
+
+		it('parses `redsocks {...}` config block', () => {
+			const redsocksConfStr = stripIndent`
+                redsocks {
+                    local_ip = 127.0.0.1;
+                    local_port = 12345;
+                    type = http-connect;
+                    ip = test2.balenadev.io;
+                    port = 1082;
+                    login = "user";
+                    password = "pass";
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({
+				redsocks: {
+					local_ip: '127.0.0.1',
+					local_port: 12345,
+					type: 'http-connect',
+					ip: 'test2.balenadev.io',
+					port: 1082,
+					login: '"user"',
+					password: '"pass"',
+				},
+			});
+		});
+
+		// Allow invalid fields through for backwards compatibility
+		it('parses to redsocks config with warnings while any values are invalid', () => {
+			const redsocksConfStr = stripIndent`
+                redsocks {
+                    local_ip = 123;
+                    local_port = foo;
+                    type = socks6;
+                    ip = 456;
+                    port = bar;
+                    login = user;
+                    password = pass;
+                }
+            `;
+			(log.warn as SinonStub).resetHistory();
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect((log.warn as SinonStub).lastCall.args[0]).to.equal(
+				'Unexpected redsocks.conf configuration detected, proxy may not behave as expected:\n' +
+					'Expecting NumericIdentifier at 0.0.port but instead got: "bar" (must be be an positive integer)\n' +
+					'Expecting one of:\n' +
+					'    "socks4"\n' +
+					'    "socks5"\n' +
+					'    "http-connect"\n' +
+					'    "http-relay"\n' +
+					'at 0.0.type but instead got: "socks6"\n' +
+					'Expecting DoubleQuotedString at 0.1.login but instead got: <REDACTED> (must be enclosed in double quotes)\n' +
```

Comment:
I think we can skip the quotes validation, if it is not double quoted, we just use the value as is

Reply:
Amended all relevant commits to remove double quoted string validation. `RedsocksConf.parse` now removes the double quotes from auth fields if they exist when parsing from the raw file string in 0ce259c1845c9137b07618aa6aafa54361771693, and `RedsocksConf.stringify` now adds the the double quotes if they don't exist around auth fields when stringifying RedsocksConfig to file string in 9e89091ce35dbf010ada7e8167c3146acaf4ca3c.